test/AutoRest.TestServer.Tests/MgmtTest/StringExtensionsTests.cs

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using NUnit.Framework;
+using AutoRest.CSharp.MgmtTest.TestCommon;
+
+namespace AutoRest.TestServer.Tests.MgmtTest
+{
+    public class StringExtensionsTests
+    {
+        [Test]
+        public void TestRefScenarioDefinedVariablesToString()
+        {
+            var scenarioDefinedVariables = new List<string> {
+                "variable1",
+                "variable2",
+            };
+            Assert.AreEqual("".RefScenarioDefinedVariablesToString(scenarioDefinedVariables), "");
+            Assert.AreEqual("{'abc': '$(variable1)'}".RefScenarioDefinedVariablesToString(scenarioDefinedVariables), "{{'abc': '{variable1}'}}");
+            Assert.AreEqual("..$(variable1).$(variable2).$(variable3).$variable2).${variable2}".RefScenarioDefinedVariablesToString(scenarioDefinedVariables), "..{variable1}.{variable2}.$(variable3).$variable2).${{variable2}}");
+            Assert.AreEqual("..$(variable1).$(variable2).$(variable3).$variable2).${variable2}".RefScenarioDefinedVariablesToString(Enumerable.Empty<string>()), "..$(variable1).$(variable2).$(variable3).$variable2).${{variable2}}");
+        }
+
+        [Test]
+        public void TestRefScenarioDefinedVariables()
+        {
+            var scenarioDefinedVariables = new List<string> {
+                "variable1",
+                "variable2",
+            };
+            Assert.AreEqual("".RefScenarioDefinedVariables(scenarioDefinedVariables).ToString(), "");
+            Assert.AreEqual("{'abc': '$(variable1)'}".RefScenarioDefinedVariables(scenarioDefinedVariables).ToString(), "${{'abc': '{variable1}'}}");
+            Assert.AreEqual("{'abc': '$(variable3)'}".RefScenarioDefinedVariables(scenarioDefinedVariables).ToString(), "{'abc': '$(variable3)'}");
+            Assert.AreEqual("{'abc': '$(variable3)'}".RefScenarioDefinedVariables(null).ToString(), "{'abc': '$(variable3)'}");
+        }
+
+        [Test]
+        public void TestToSnakeCase()
+        {
+            Assert.AreEqual("".ToSnakeCase(), "");
+            Assert.AreEqual("I".ToSnakeCase(), "i");
+            Assert.AreEqual("IAmHere".ToSnakeCase(), "i_am_here");
+            Assert.AreEqual("iAmHere".ToSnakeCase(), "i_am_here");
+        }
```

Comment:
The test case here is not quite the way we use in `NUnit`
```suggestion
        [TestCase("", "")]
        [TestCase("I", "i")]
        [TestCase("IAmHere", "i_am_here")]
        [TestCase("iAmHere", "i_am_here")]
        public void TestToSnakeCase(string input, string expected)
        {
            var result = input.ToSnakeCase();
            Assert.AreEqual(expected, result);
        }
```
This comment applies to other test case as well.